src/commands/deploy/remix/deploy.ts

```diff
@@ -0,0 +1,280 @@
+        : cwd;
+
+    // Prepare services before deploying (database, authentication, etc)
+    await prepareServicesPreBackendDeployment(
+        genezioConfig,
+        genezioConfig.name,
+        options.stage,
+        options.env,
+    );
+
+    // Install dependencies
+    const installDependenciesCommand = await attemptToInstallDependencies(
+        [],
+        componentPath,
+        packageManagerType,
+    );
+
+    // Check if the project uses remix vite by looking for vite.config.* files
+    const isRemixVite =
+        fs.existsSync(path.join(componentPath, "vite.config.ts")) ||
+        fs.existsSync(path.join(componentPath, "vite.config.js")) ||
+        fs.existsSync(path.join(componentPath, "vite.config.mjs")) ||
+        fs.existsSync(path.join(componentPath, "vite.config.cjs"));
+
+    if (isRemixVite) {
+        debugLogger.debug("Building the project using: remix vite:build");
+        // Build the project using remix vite
+        await $({
+            stdio: "inherit",
+            cwd: componentPath,
+        })`remix vite:build`.catch(() => {
+            throw new UserError("Failed to build the Remix project. Check the logs above.");
+        });
+    } else {
+        debugLogger.debug("Building the project using: npx remix build");
+        // Build the project using remix build
+        await $({
+            stdio: "inherit",
+            cwd: componentPath,
+        })`npx remix build`.catch(() => {
+            throw new UserError("Failed to build the Remix project. Check the logs above.");
+        });
+    }
+
+    const remixBuildCommand = isRemixVite ? "remix vite:build" : "npx remix build";
+
+    // Add remix component to config
+    await addSSRComponentToConfig(
+        options.config,
+        {
+            path: componentPath,
+            packageManager: packageManagerType,
+            scripts: {
+                build: [remixBuildCommand],
+                deploy: [`${installDependenciesCommand.command}`, remixBuildCommand],
+            },
+        },
+        SSRFrameworkComponentType.remix,
+    );
+
+    // Copy the build folder to /tmp
+    const tempBuildCwd = await createTemporaryFolder();
+    debugLogger.debug(`Copying project files to ${tempBuildCwd}`);
+
+    // Check and copy build folder
+    const buildPath = path.join(componentPath, "build");
+    await fs.promises
+        .cp(buildPath, tempBuildCwd, { recursive: true, force: true, dereference: true })
+        .catch(() => {
+            throw new UserError("Failed to copy project build folder to temporary directory");
+        });
+
+    if (!isRemixVite) {
+        // Copy public folder
+        const publicPath = path.join(componentPath, "public");
+        await fs.promises
+            .mkdir(path.join(tempBuildCwd, "public"), { recursive: true })
+            .catch(() => {
+                throw new UserError("Failed to create public folder in temporary directory");
```

Comment:
We are not displaying the root cause error. It's hard as a user to understand what I can do actionable after receiving this error.